pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/services/ReportMutationService.kt

```diff
@@ -63,277 +88,116 @@ class ReportMutationService : KoinComponent {
+                    schemaFileNames = exception.schemaFileNames.toList()
+                )
+            }
 
-            when (upsertAction) {
-                UpsertAction.CREATE -> createReport(report)
-                UpsertAction.REPLACE -> replaceReport(report)
+            else -> {
+                return UpsertReportResult(
+                    result = "FAILURE",
+                    reason = result.exceptionOrNull()?.message ?: "unknown"
+                )
             }
-        } catch (e: BadRequestException) {
-            logger.error("Validation error during upsert: ${e.message}", e)
-            throw e // Re-throwing the BadRequestException
-        } catch (e: ContentException) {
-            logger.error("Content error during upsert: ${e.message}", e)
-            throw e // Re-throwing the ContentException
-        } catch (e: Exception) {
-            logger.error("Unexpected error during upsert: ${e.message}", e)
-            throw ContentException("Failed to perform upsert: ${e.message}")
         }
     }
 
-    /**
-     * Maps the given ReportInput to a Report object.
-     *
-     * This method extracts the necessary fields from the input and constructs a Report instance.
-     * It also parses the content based on its type.
-     *
-     * @param input The ReportInput containing the details to map to a Report.
-     * @return A Report object populated with data from the input.
-     */
-    @Throws(ContentException::class)
-    private fun mapInputToReport(input: ReportInput): Report {
-
-        return try {
-
-            // Parse the content based on its type
-            val parsedContent = input.content?.let { parseContent(it, input.contentType) } as? Map<*, *>?
-
-            // Set id and reportId to be the same
-            val reportId = input.id // Generate a new ID if not provided
-
-
-            Report(
-                id = reportId,
-                uploadId = input.uploadId,
-                reportId = reportId, //Set reportId to be the same as id
-                dataStreamId = input.dataStreamId,
-                dataStreamRoute = input.dataStreamRoute,
-                dexIngestDateTime = input.dexIngestDateTime,
-                messageMetadata = input.messageMetadata?.let { mapInputToMessageMetadata(it) },
-                stageInfo = input.stageInfo?.let { mapInputToStageInfo(it) },
-                tags = input.tags?.associate { it.key to it.value },
-                data = input.data?.associate { it.key to it.value },
-                contentType = input.contentType,
-                jurisdiction = input.jurisdiction,
-                senderId = input.senderId,
-                dataProducerId = input.dataProducerId,
-                content = parsedContent,
-                timestamp = input.timestamp
+    private fun validateReport(input: Map<String, Any?>?): ValidatedReportResult {
+        try {
+            logger.info("The report received: $input and will be converted to Json")
```

Comment:
Should we check whether input is null, here?

Reply:
Added null or empty check on the report input.